test/e2e/runner/rpc.go

```diff
@@ -17,21 +17,47 @@ import (
 // progress at all.
 func waitForHeight(testnet *e2e.Testnet, height int64) (*types.Block, *types.BlockID, error) {
 	var (
-		err          error
-		maxResult    *rpctypes.ResultBlock
-		clients      = map[string]*rpchttp.HTTP{}
-		lastIncrease = time.Now()
+		err             error
+		maxResult       *rpctypes.ResultBlock
+		clients         = map[string]*rpchttp.HTTP{}
+		lastIncrease    = time.Now()
+		nodesAtHeight   = map[string]struct{}{}
+		numRunningNodes int
 	)
+	for _, node := range testnet.Nodes {
+		if node.Mode == e2e.ModeSeed {
+			continue
+		}
+
+		if node.Mode == e2e.ModeLight {
+			continue
+		}
+
+		if node.HasStarted {
+			numRunningNodes++
+		}
+	}
 
 	for {
 		for _, node := range testnet.Nodes {
+			// skip nodes that have reached the target height
+			if _, ok := nodesAtHeight[node.Name]; ok {
+				continue
+			}
+
 			if node.Mode == e2e.ModeSeed {
 				continue
 			}
+
 			if node.Mode == e2e.ModeLight {
 				continue
 			}
 
+			if !node.HasStarted {
+				continue
+			}
```

Comment:
nit: we could turn this into a method for `Node` (I'm not sure if we use this combination elsewhere but it seems likely)